packages/matchbox/src/components/Tooltip/TooltipOverlay.js

```diff
@@ -1,76 +1,75 @@
-import React, { Component, Fragment } from 'react';
+import React from 'react';
 import PropTypes from 'prop-types';
 import { Portal } from '../Portal';
 import { WindowEvent } from '../WindowEvent';
 import { debounce } from '../../helpers/event';
 import { getPositionFor, getPreferredDirectionFor } from '../../helpers/geometry';
-import styles from './TooltipOverlay.module.scss';
+import { tokens } from '@sparkpost/design-tokens';
+import { Box } from '../Box';
 
-class TooltipOverlay extends Component {
-  static displayName = 'TooltipOverlay';
+const defaultPosition = {
+  top: 0,
+  left: 0,
+  width: 0,
+  height: 0,
+};
 
-  static defaultProps = {
-    eventDebounce: 400
-  }
+const defaultDirection = {
+  top: null,
+  bottom: null,
+  left: null,
+  right: null,
+};
 
-  static propTypes = {
-    renderActivator: PropTypes.func.isRequired,
-    renderTooltip: PropTypes.func.isRequired,
-    eventDebounce: PropTypes.number
-  }
+function TooltipOverlay(props) {
+  const [position, setPosition] = React.useState(defaultPosition);
+  const [preferredDirection, setPreferredDirection] = React.useState(defaultDirection);
+  const activatorRef = React.useRef(null);
 
-  state = {
-    position: {
-      top: 0,
-      left: 0,
-      width: 0,
-      height: 0
-    },
-    preferredDirection: {
-      top: null,
-      bottom: null,
-      left: null,
-      right: null
-    }
-  }
+  const { renderTooltip, renderActivator, eventDebounce, portalId } = props;
 
-  componentDidMount() {
-    this.handleMeasurement();
+  function handleMeasurement() {
+    if (activatorRef.current) {
+      setPosition(getPositionFor(activatorRef.current));
+      setPreferredDirection(getPreferredDirectionFor(activatorRef.current));
+    }
   }
 
-  componentWillReceiveProps() {
-    this.handleMeasurement();
-  }
+  React.useLayoutEffect(() => {
+    handleMeasurement();
+  }, [renderTooltip, renderActivator, activatorRef]);
 
-  handleMeasurement = () => {
-    this.setState({
-      preferredDirection: getPreferredDirectionFor(this.activator),
-      position: getPositionFor(this.activator)
-    });
-  }
+  return (
+    <>
+      <WindowEvent event="resize" handler={debounce(handleMeasurement, eventDebounce)} />
+      <WindowEvent event="scroll" handler={debounce(handleMeasurement, eventDebounce)} />
```

Comment:
I know this isn't new, but do we need to call handleMeasurement on resize and scroll events?  The side effect already handles mouseover and mouseout.

Reply:
Resize can be removed, but now that we are using fixed positioning, this is the only thing that will hide tooltips if you one is visible before you start scrolling